### ISSUE TITLE
Apple M1 Chip not supported in offline bootstrapper

### ISSUE THREAD

**Issue by boonhapus** [thoughtspot/cs_tools#29]
### First Stop

- [X] I have read and agreed to the ThoughtSpot Dev Tools EULA.
- [X] I understand that if I have a critical issue with ThoughtSpot, I should reach out to my account team instead of cs_tools.
- [X] I searched the cs_tools documentation, with the integrated search.

### Platform Configuration

```
    [PLATFORM DETAILS]
    system: Darwin (detail: macOS-12.3.1-arm64-arm-64bit)
    platform tag 'macosx-10.9-universal2'
    python: 3.9.13
    ran at: 2022-07-05 14:30:14
```

### Example Command

`python3 bootstrap --install`

### Description

```
INFO     | 2022-07-05 14:30:18,258 | _activator.py:195 | Installing requirements via pip.
WARNING  | 2022-07-05 14:30:18,583 | _activator.py:163 | An error occurred. Removing partial environment.
ERROR    | 2022-07-05 14:30:18,783 | _main.py:88 | CS Tools installation failed.
DEBUG    | 2022-07-05 14:30:18,783 | _main.py:89 | full traceback..
Reason: Looking in links: /Users/namey.namerson/Downloads/cs_tools-bootstrapper/pkgs
Ignoring async-generator: markers 'python_version < "3.7" and python_version >= "3.6"' don't match your environment
Ignoring colorama: markers 'python_version >= "3.6" and python_full_version < "3.0.0" and platform_system == "Windows" or platform_system == "Windows" and python_version >= "3.6" and python_full_version >= "3.5.0"' don't match your environment
Ignoring contextlib2: markers 'python_version < "3.7" and python_version >= "3.6" and python_full_version >= "3.6.1"' don't match your environment
Ignoring contextvars: markers 'python_version < "3.7" and python_version >= "3.6" and python_full_version >= "3.6.2"' don't match your environment
Ignoring dataclasses: markers 'python_version >= "3.6" and python_version < "3.7" and python_full_version >= "3.6.3" and python_full_version < "4.0.0"' don't match your environment
Ignoring greenlet: markers 'python_version >= "3" and (platform_machine == "aarch64" or platform_machine == "ppc64le" or platform_machine == "x86_64" or platform_machine == "amd64" or platform_machine == "AMD64" or platform_machine == "win32" or platform_machine == "WIN32") and python_full_version >= "3.6.1" and python_full_version < "4.0.0"' don't match your environment
Ignoring immutables: markers 'python_version < "3.7" and python_version >= "3.6" and python_full_version >= "3.6.2"' don't match your environment
Ignoring importlib-metadata: markers 'python_version >= "3.6" and python_version < "3.8" and python_full_version >= "3.6.1" and python_full_version < "4.0.0"' don't match your environment
Ignoring pyreadline3: markers 'sys_platform == "win32"' don't match your environment
Ignoring zipp: markers 'python_version >= "3.6" and python_version < "3.8"' don't match your environment
Processing ./pkgs/anyio-3.6.1-py3-none-any.whl
Processing ./pkgs/asgiref-3.4.1-py3-none-any.whl
Processing ./pkgs/certifi-2022.6.15-py3-none-any.whl
Processing ./pkgs/charset_normalizer-2.1.0-py3-none-any.whl
Processing ./pkgs/click-8.0.4-py3-none-any.whl
Processing ./pkgs/commonmark-0.9.1-py2.py3-none-any.whl
Processing ./pkgs/fastapi-0.78.0-py3-none-any.whl
Processing ./pkgs/h11-0.12.0-py3-none-any.whl
Processing ./pkgs/httpcore-0.14.7-py3-none-any.whl
Processing ./pkgs/httpx-0.22.0-py3-none-any.whl
Processing ./pkgs/idna-3.3-py3-none-any.whl
Processing ./pkgs/Jinja2-3.0.3-py3-none-any.whl
ERROR: Could not find a version that satisfies the requirement markupsafe==2.0.1 (from versions: none)
ERROR: No matching distribution found for markupsafe==2.0.1
---

Traceback (most recent call last):
  File "/Users/namey.namerson/Downloads/cs_tools-bootstrapper/bootstrap/_main.py", line 72, in run
    rc = activator.run()
  File "/Users/namey.namerson/Downloads/cs_tools-bootstrapper/bootstrap/_activator.py", line 66, in run
    self._install(remote)
  File "/Users/namey.namerson/Downloads/cs_tools-bootstrapper/bootstrap/_activator.py", line 139, in _install
    self._install_cs_tools(env)
  File "/Users/namey.namerson/Downloads/cs_tools-bootstrapper/bootstrap/_activator.py", line 196, in _install_cs_tools
    cp = env.pip("install", "-r", (PKGS / "requirements.txt").as_posix(), *common)
  File "/Users/namey.namerson/Downloads/cs_tools-bootstrapper/bootstrap/_virtual_env.py", line 69, in pip
    return self.python("-m", "pip", "--isolated", *args, **kwargs)
  File "/Users/namey.namerson/Downloads/cs_tools-bootstrapper/bootstrap/_virtual_env.py", line 66, in python
    return self.run(str(self.exe), *args, **kwargs)
  File "/Users/namey.namerson/Downloads/cs_tools-bootstrapper/bootstrap/_virtual_env.py", line 60, in run
    raise CSToolsActivatorError(return_code=cp.returncode, log=cp.stdout.decode())
_errors.CSToolsActivatorError
```

**Comment by boonhapus** [comment]
The offline packages are built against the most common architecture: `x86_64`. Apple's new M1 chip is an `ARM64` based architecture. We'll need to add this to the build process in nox.

**Comment by boonhapus** [comment]
For installs with internet connectivity that can reach pypi, you can run the following steps.

```bash
$ source "$HOME/Library/Application Support/cs_tools/.cs_tools/bin/activate"
(.cs_tools) $ pip install -r $HOME/Downloads/cs_tools-bootstrapper/pkgs/requirements.txt
(.cs_tools) $ deactivate
$ python bootstrap --install
```


For offline installs, there currently is no workaround.

**Comment by boonhapus** [comment]
This is resolved as part of V1.4.0 . Check the release notes to learn more.

https://thoughtspot.github.io/cs_tools/changelog/1.4.0
